packages/server/computation_container/client/computation_to_db/client.hpp

```diff
@@ -89,5 +84,27 @@ class Client final
         writer.write();
         std::ofstream(resultDbPath + job_uuid + "/completed");
     }
+
+    // TableWriterとComputationResultWriterは本質的には同じことをしているが，
+    // 微妙に形式が異なるため別の実装としている
+    // 例)send_share時は行ごとにpieceとするが，計算結果は値ごとにpieceとしている
+    class TableWriter
```

Comment:
i think if inner class don't use outer class, it would like to be separate another file.

Reply:
I want to make it inner class because `TableWriter` uses `Client::shareDbPath`.

Reply:
i see.
but path is static object . right ?
if the tablewriter wants path strings ,then  it would be given input arguments.
or maybe Client::dbPath to get the directory?

Reply:
Yes, It is a static object, but it is private.
The path is private because I don't want it to affect anything outside the client, but would it be better to make it public or accept it as an argument?

Reply:
ah, I understand!
Are you saying that TableWriter should only be responsible for piece division? Without writing directly to the DB. This certainly looks better.
I have tried to correct it below, but please check if it is consistent with your intention.
5da8e19228bef85bf773d1f9270138a44f3645e7

Reply:
its correct!